src/drivers/uavcan/sensors/battery.cpp

```diff
@@ -43,30 +43,102 @@ UavcanBatteryBridge::UavcanBatteryBridge(uavcan::INode &node) :
 	ModuleParams(nullptr),
 	_sub_battery(node),
 	_sub_battery_aux(node),
+	_pub_battery_info(node),
+	_pub_battery_info_aux(node),
 	_warning(battery_status_s::BATTERY_WARNING_NONE),
 	_last_timestamp(0)
 {
 }
 
 int UavcanBatteryBridge::init()
 {
-	int res = _sub_battery.start(BatteryInfoCbBinder(this, &UavcanBatteryBridge::battery_sub_cb));
 
-	if (res < 0) {
-		PX4_ERR("failed to start uavcan sub: %d", res);
-		return res;
-	}
+	int32_t uavcan_sub_bat = 0;
+	param_get(param_find("UAVCAN_SUB_BAT"), &uavcan_sub_bat);
+
+	if (uavcan_sub_bat) {
+		int res = _sub_battery.start(BatteryInfoCbBinder(this, &UavcanBatteryBridge::battery_sub_cb));
+
+		if (res < 0) {
+			PX4_ERR("failed to start uavcan sub: %d", res);
+			return res;
+		}
 
-	res = _sub_battery_aux.start(BatteryInfoAuxCbBinder(this, &UavcanBatteryBridge::battery_aux_sub_cb));
+		res = _sub_battery_aux.start(BatteryInfoAuxCbBinder(this, &UavcanBatteryBridge::battery_aux_sub_cb));
 
-	if (res < 0) {
-		PX4_ERR("failed to start uavcan sub: %d", res);
-		return res;
+		if (res < 0) {
+			PX4_ERR("failed to start uavcan sub: %d", res);
+			return res;
+		}
```

Comment:
Any specific reason we want to check this parameter here? The UavcanBatteryBridge won't be created in sensor_bridge.cpp unless the parameter is true.

Reply:
The bridge is bidirectional, so will be created even if just UAVCAN->uORB is enabled. This check is for uORB->UAVCAN